src/semantic/type_checker.rs

```diff
@@ -0,0 +1,161 @@
+#![allow(unused_variables)]
+use lexical::types::Types;
+use syntax::expr::ExprWrapper;
+use syntax::expr::Expr::*;
+use semantic::analyzer_trait::ASTAnalyzer;
+
+pub struct TypeChecker;
+
+impl TypeChecker {
+    pub fn new() -> TypeChecker {
+        TypeChecker
+    }
+
+    fn cmp_lhs_rhs(lhs_str: String, rhs_str: String) -> Option<String> {
```

Comment:
I don't think you can impl anything for specific enum variants. Only the enum itself. 

And it's doing a string compare because the type could be a non builtin type. IE MyClassFoo
If they're the same: int vs int, or MyClassFoo vs MyClassFoo, both are valid. But one is a builtin and the later is not